lms/djangoapps/grades/tasks.py

```diff
@@ -328,3 +347,13 @@ def _course_task_args(course_key, **kwargs):
 
     for offset in six.moves.range(0, enrollment_count, batch_size):
         yield (six.text_type(course_key), offset, batch_size)
+
+
+def _are_grades_frozen(course_key):
```

Comment:
We may need to add this check in `handle_grading_policy_changed` signal handler as well?

Reply:
It's not necessary since that handler simply fires the `compute_all_grades_for_course` task above.

As mentioned earlier, with persistent grades enabled, grades are _only_ updated by these celery tasks.  So this is a fail-safe place to add this check.  All other signal handlers/etc throughout the code will forward to these 4 tasks.